src/rpm/package.rs

```diff
@@ -148,34 +156,54 @@ impl RPMPackage {
     #[cfg(feature = "signature-meta")]
     pub fn verify_signature<V>(&self, verifier: V) -> Result<(), RPMError>
     where
```

Comment:
There are a few open questions:

* If you attempt to verify the signatures of a package with no signatures, what error message should it return?  `RPMError::NoSignatureFound` is used for something different, but the signature of `RPMError::VerificationError`  is designed in such a way that it can't easily be used outside of `Verifier` unless we make a few more changes to that API.  New error type, or adjust the existing ones?

* Lets say the package is EdDSA signed and I attempt to verify using a Verifier built from an RSA public key.  It ought to fail (but currently it doesn't).  But again,  `RPMError::VerificationError`  can't be used easily outside of `Verifier`.  I suppose that we could force it to try to verify the signature even though we know it will fail, which would raise the error we expect?

Reply:
@drahnr Any feedback?